fastlane_bot/events/exchanges/solidly_v2.py

```diff
@@ -57,17 +57,70 @@ async def _get_tkn0_B(contract: Contract) -> str:
 async def _get_tkn1_B(contract: Contract) -> str:
     return "0xe5D7C2a44FfDDf6b295A15c148167daaAf5Cf34f" # TODO Use the constant WRAPPED_GAS_TOKEN_ADDRESS for this network
 
+
+def _get_pool_function_1(factory_contract):
+    """ Function to get pools from Factory.
+    This function is intended to be used with a Multicall. It fetches pools from a Solidly fork Factory contract.
+
+    Args:
+        factory_contract: The factory contract.
+
+    Returns:
+        The function.
+
+    """
+    return factory_contract.functions.getPair
+def _get_pool_function_2(factory_contract):
+    """ Function to get pools from Factory.
+        This function is intended to be used with a Multicall. It fetches pools from a Solidly fork Factory contract.
+
+        Args:
+            factory_contract: The factory contract.
+
+        Returns:
+            The function.
+
+        """
+    return factory_contract.functions.getPool
+
+
+def _get_pool_args_1(tkn0, tkn1, stable):
+    """ Function to manage args input to get pools from Factory.
+
+    Args:
+        tkn0: The first token address.
+        tkn1: The second token address.
+        stable: (bool) If True, indicates a stable pool. If False, indicates a Volatile pool.
+    Returns:
+        The function returns the arguments necessary to get pool addresses from the factory contract.
+
+    """
+    return tkn0, tkn1, stable
+
+def _get_pool_args_2(tkn0, tkn1, stable):
+    """ Function to manage args input to get pools from Factory.
+
+    Args:
+        tkn0: The first token address.
+        tkn1: The second token address.
+        stable: (bool) If True, indicates a stable pool. If False, indicates a Volatile pool.
+    Returns:
+        The function returns the arguments necessary to get pool addresses from the factory contract.
+
+    """
+    return tkn0
```

Comment:
It probably needs to be `return [tkn0]` in order for it to work when used.

Reply:
Not relevant any more